testing/trino-tests/src/test/java/io/trino/tests/TestAdaptivePartialAggregation.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.testing.AbstractTestAggregations;
+import io.trino.testing.QueryRunner;
+import io.trino.tests.tpch.TpchQueryRunnerBuilder;
+
+public class TestAdaptivePartialAggregation
+        extends AbstractTestAggregations
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        return TpchQueryRunnerBuilder.builder()
+                .setExtraProperties(ImmutableMap.of(
+                        "adaptive-partial-aggregation.min-rows", "1000"))
```

Comment:
inline with previous line

set it to `0`?

you should also reduce PA buffer to `0` so that flush happens after every page

Reply:
yeah, `min-rows=0 `makes sense.
I would leave `task.max-partial-aggregation-memory` with default as flushing per page is not a realistic scenario and there should be enough splits in the test data for flush per split to use adaptation.

Reply:
ok, I added `task.max-partial-aggregation-memory=0B`